src/hooks/useDrawerLlmChat.ts

```diff
@@ -0,0 +1,396 @@
+  LlmHealthCheck,
+  LlmMessage,
+  UseChatMessagesReturn,
+  UseFileAttachmentsReturn,
+  UseLlmServiceReturn,
+  UseTextareaResizeReturn,
+} from '@/types';
+
+/**
+ * Configuration constants for the chat drawer
+ */
+export const chatConfig = {
+  maxFileSize: 10 * 1024 * 1024,
+  maxTextAreaHeight: 200,
+  minTextAreaHeight: 40,
+  requestTimeout: 15000,
+  temperature: 0.7,
+} as const;
+
+/**
+ * Allowed file types for attachment
+ */
+const allowedFileTypes: string[] = [
+  'image/jpeg',
+  'image/png',
+  'image/gif',
+  'image/webp',
+  'text/plain',
+  'application/pdf',
+  'application/msword',
+  'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+  'application/json',
+  'text/csv',
+] as const;
+
+/**
+ * Custom hook for managing chat messages
+ *
+ * Provides functionality for managing the chat message state including
+ * adding messages, updating messages, and generating unique IDs.
+ *
+ * @returns Object with message management functions
+ */
+export const useChatMessages = (): UseChatMessagesReturn => {
+  const [messages, setMessages] = useState<ChatMessage[]>([]);
+
+  /**
+   * Generates a unique message ID
+   * @returns Unique message identifier
+   */
+  const generateMessageId = useCallback((): string => {
+    return `msg-${uuidv4()}`;
+  }, []);
+
+  /**
+   * Adds a new message to the chat
+   * @param message - Message to add
+   */
+  const addMessage = useCallback((message: ChatMessage) => {
+    setMessages((prev) => [...prev, message]);
+  }, []);
+
+  /**
+   * Adds multiple messages to the chat
+   * @param newMessages - Messages to add
+   */
+  const addMessages = useCallback((newMessages: ChatMessage[]) => {
+    setMessages((prev) => [...prev, ...newMessages]);
+  }, []);
+
+  /**
+   * Updates the last message in the chat
+   * @param updater - Function to update the message
+   */
+  const updateLastMessage = useCallback((updater: (message: ChatMessage) => ChatMessage) => {
+    setMessages((prev) => {
+      if (prev.length === 0) {
+        return prev;
+      }
+      const updated = [...prev];
+      updated[updated.length - 1] = updater(updated[updated.length - 1]);
+      return updated;
+    });
+  }, []);
+
+  return {
+    messages,
+    setMessages,
+    generateMessageId,
+    addMessage,
+    addMessages,
+    updateLastMessage,
+  };
+};
+
+/**
+ * Custom hook for managing file attachments
+ *
+ * Handles file upload, validation, formatting, and management.
+ * Supports various file types including images, documents, and text files.
+ *
+ * @returns Object with file management functions
+ */
+export const useFileAttachments = (): UseFileAttachmentsReturn => {
+  const [attachedFiles, setAttachedFiles] = useState<AttachedFile[]>([]);
+
+  /**
+   * Formats file size in human-readable format
+   * @param fileSizeInBytes - File size in bytes
+   * @returns Formatted file size string
+   */
+  const formatFileSize = useCallback((fileSizeInBytes: number): string => {
```

Comment:
can be replaced with `formattedValueToString(getValueFormat('decbytes')(size))` by `@grafana/data`